storage/innobase/fil/fil0fil.cc

```diff
@@ -5470,18 +5470,33 @@ fil_io_set_encryption(
 	const page_id_t&	page_id,
 	fil_space_t*		space)
 {
-	/* Don't encrypt the log, page 0 of all tablespaces, all pages
-	from the system tablespace. */
-	if (!req_type.is_log() && page_id.page_no() > 0
-	    && space->encryption_type != Encryption::NONE)
-	{
-		req_type.encryption_key(space->encryption_key,
-					space->encryption_klen,
-					space->encryption_iv);
-		req_type.encryption_algorithm(Encryption::AES);
-	} else {
+	/* Don't encrypt page 0 of all tablespaces except, don't encrypt
+	redo log and system tablespace. */
+	if (space->encryption_type == Encryption::NONE
+	    || (page_id.page_no() == 0 && !req_type.is_log())) {
+		req_type.clear_encrypted();
+		return;
+	}
+
+	/* For writting redo log, if encryption for redo log is disabled,
```

Comment:
writing

Reply:
I copied  this spelling from MySQL 8 I guess... Fixing